systemverilog-plugin/UhdmAst.cc

```diff
@@ -1836,7 +1842,32 @@ void UhdmAst::simplify_parameter(AST::AstNode *parameter, AST::AstNode *module_n
         });
     }
     // first apply custom simplification step if needed
-    simplify(parameter, nullptr);
+    simplify(parameter, module_node);
+    // workaround for yosys not simplifying parameters children sometimes
+    // parameters should have 2 children:
```

Comment:
> We assume each parameter should have 2 children, yet it seems to fail silently if `size != 2`.

Could you please explain it a bit more or provide a test case when it fails silently? I believe parameter with only 1 child should be valid.

Reply:
What I mean is that we have a comment here: https://github.com/antmicro/yosys-f4pga-plugins/blob/kr/fix_param_const_simplify/systemverilog-plugin/UhdmAst.cc#L1847
which says that there should always be 2 children.

And then there are two if statements : https://github.com/antmicro/yosys-f4pga-plugins/blob/kr/fix_param_const_simplify/systemverilog-plugin/UhdmAst.cc#L1850
https://github.com/antmicro/yosys-f4pga-plugins/blob/kr/fix_param_const_simplify/systemverilog-plugin/UhdmAst.cc#L1861
Which check for `size>0` and `size>1`.
But there is no case where it would fail because of wrong number of children.

So I'm a bit confused about the comment linked above, do we want to ensure that the children exist, or accept any case, even with 0 children or more than 2?

Reply:
Oh, now I get what you mean. Yes, the comment is a bit misleading there. I'll change it and add assert to make sure that parameter have at least 1 child (I don't think parameter node without any child is valid).